Simulator/simulation/simulator.js

```diff
@@ -2,17 +2,63 @@ const fs = require('fs');
 const House = require('./house');
 
 Simulator = {
-	runSimulation: function() {
-		const houses = [];
-		const config = JSON.parse(fs.readFileSync('config/house.json', 'utf8'));
+	price: 0,
+	power: 1000,
+	houses: [],
+	powerGain: 0,
+	powerLoss: 0,
+
+	/**
+	 * Sell power to a house
+	 * @param {number} powerNeed
+	 * @return {number}
+	 */
+	sellPower: function(powerNeed) {
+		this.power -= powerNeed;
+		this.powerLoss += powerNeed;
+
+		// prevent negative values until buying from a coal plant is possible
+		if (this.power < 0) this.power = 0;
+
+		// TODO: buy power from a power plant if needed
+		return powerNeed;
+	},
+
+	/**
+	 * Buy power from a house
+	 * @param {number} power
+	 */
+	buyPower: function(power) {
+		this.power += power;
+		this.powerGain += power;
+	},
+
+	initHouses: function(configPath) {
+		const config = JSON.parse(fs.readFileSync(configPath, 'utf8'));
 
 		config.forEach((initValues) => {
-			houses.push(new House(initValues));
+			this.houses.push(new House(initValues, this));
 		});
+	},
+
+	runSimulation: function() {
+		this.initHouses('config/house.json');
 
-		houses.forEach((house) => {
-			house.runSimulation();
+		this.houses.forEach((house) => {
+			setInterval(house.runSimulation, house.pollTime, house);
 		});
+		setInterval(this.calcPrice, 1000, this);
+	},
+	calcPrice: function(self) {
```

Comment:
Should we have different polls for the houses and simulation or use the same?
Eg:
```javascript
runSimulation: function()
    this.houses.forEach((house) => {
        setInterval(house.runSimulation, house.pollTime, house);
    });
    setInterval(this.calcPrice, 1000, this);
},
```
Or:
```javascript
setInterval(this.runSimulation, 1000);

runSimulation: function()
    this.houses.forEach((house) => {
	 house.runSimulation();
    });
    this.calcPrice();
},
```